searx/enginelib/traits.py

```diff
@@ -24,6 +23,9 @@
     from . import Engine
 
 
+Self = TypeVar("Self", bound="EngineTraits")  # Python 3.11, to replace by typing.Self
+
```

Comment:
> Seems ok even with Python 3.8

Nope .. the pyright we use in the CI is not based on the python version installed in the CI /BTW we ignore issues from pyright in the CI.

IDK how to test in python 3.8 .. but in my IDE I see that "Self" replacement does not work .. it will be used for `EngineTraits` class and `EngineTraitsMap` in the code below.

Give me a few minutes, I prepare a WIP commit with a [forward references](https://peps.python.org/pep-0484/#forward-references) as mentioned in my comment https://github.com/searxng/searxng/pull/3260#issuecomment-1973280996

Reply:
Simpler with the class name instead of Self

Reply:
> Simpler with the class name instead of Self

OK, feel fre to squash the WIP commit into you initial commit .. after we can merge the PR.